app/queries/get_course_option_from_codes.rb

```diff
@@ -66,4 +57,26 @@ def initialize(
   def call
     course_option if valid?
   end
+
+  def self.get_unique_course_option(record, attr)
+    course_option_attrs = { study_mode: record.study_mode }
+
+    course_option_attrs[:site] = record.site if record.site
+
+    possible_course_options = record.course.course_options.where(course_option_attrs)
+
+    if possible_course_options.count == 1
+      record.course_option ||= possible_course_options.first
+    else
+      record.errors.add(attr, build_course_option_error_message(record, possible_course_options.count))
+    end
+  end
+
+  def self.build_course_option_error_message(record, match_count)
+    error_message = match_count.zero? ? 'cannot find' : 'cannot find unique'
+
+    error_message << " #{record.study_mode} option"
```

Comment:
I realised that we will never see this error due to the controller: https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/controllers/vendor_api/decisions_controller.rb#L10 which just returns a generic error if it is missing. I reckon we should amend this to return the expected errors